### ISSUE TITLE
Linux 6.6.39 fails to mount /boot and setting up virtual console, freezing on startup

### ISSUE THREAD

**Issue by IncredibleLaser** [NixOS/nixpkgs#327048]
### Describe the bug
When trying to boot the system using Linux 6.6.39 (current unstable), the message on screen states "Mounting /boot..." and "Setting up Virtual Console...", eventually freezing after a couple of seconds. The system must be hard reset.

### Steps To Reproduce
1. Update to Linux 6.6.39
2. Boot into new system

### Expected behavior
A working boot that presents me with a login screen

### Additional context
The problem is neither exhibited using Linux 6.6.37 (previous LTS in unstable) in nor with 6.9.9 (current -latest). My /boot is a FAT32 formatted partition marked as EFI BOOT.

### Notify maintainers
@TredwellGit @Ma27 @NeQuissimus @alyssais @thoughtpolice 

### Metadata

```
laser@powerbox ~> nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.9.9, NixOS, 24.11 (Vicuna), 24.11.20240712.7e7c39e`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.5`
 - nixpkgs: `/nix/store/qmh8bas1qni03drm0lnjas2azh7h87cn-source`
```

However, please note that this is with the working kernel as the command cannot be run when the bug is encountered.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by alyssais** [comment]
There's not really much that the Nixpkgs kernel maintainers can do about this sort of thing, as our kernels are basically unmodified from upstream.  If you want to make progress on this, you should bisect the kernel and take it upstream.

**Comment by alyssais** [comment]
6.6.40 is out, and it sounds like it fixes an issue that could be this, so maybe try that?

**Comment by alyssais** [comment]
https://bugzilla.kernel.org/show_bug.cgi?id=219039

**Comment by IncredibleLaser** [comment]
Thanks for looking into it, unfortunately I can't test until Friday as I'm traveling. I'll try as soon as I can and close the issue if it's fixed!

**Comment by IncredibleLaser** [comment]
Fixed in 6.6.40
